src/modules/mentorships/__tests__/mentorships.controller.spec.ts

```diff
@@ -125,13 +129,37 @@ describe('modules/mentorships/MentorshipsController', () => {
       ).rejects.toThrow(BadRequestException);
     });
 
+    it(`should return a 400 error if a mentee requested more mentorships than limit`, async () => {
+      const mentorships = [];
+      for (let i = 0; i < Config.maximumOpenMentorships; i++) {
+        mentorships.push(<Mentorship>{ _id: new ObjectIdMock(i.toString()) });
+      }
+      mentorshipsService.getMenteeMentorshipsByStatus = jest.fn(() =>
+        Promise.resolve(mentorships),
+      );
+      const expectedException = new BadRequestException(
+        `mentees pending/open mentorships request are limited to 5`,
+      );
+      expect.assertions(1);
+      try {
+        await mentorshipsController.applyForMentorship(
+          request,
+          '123',
+          mentorship,
+        );
+      } catch (e) {
+        expect(e.message).toStrictEqual(expectedException.message);
```

Comment:
Is there a better in Typescript to assert over the `expectedException` object using 

```
await expect(
         mentorshipsController.applyForMentorship(request,'123',mentorship);
      ).rejects.toThrow(expectedException); // i'd like to use this sytax to assert over the BadRequestException error message 
```

Reply:
Something like that?

https://github.com/Coding-Coach/find-a-mentor-api//blob/80b1e6dfdcd394a3fdcd497746baaca7a4969311/src/modules/admin/__tests__/mentors.controller.spec.ts#L208-L210

Reply:
I was looking for a way in TS to inspect the exception object. As the endpoint returns multiple BadRequests I wanted to test that I was triggering the right BadRequest. If it can help, for comparison, in python   I would have done something like :

```python
def  test_endpoint_sad_path(self):  # BadRequestException is the exception class
        with self.assertRaises(BadRequestException) as ctx:
                my_function_which_raises()
        self.assertEqual('are you trying to mentor yourself', ctx.exception)
```